Sources_v3/Controllers/ChannelController+SwiftUI.swift

```diff
@@ -0,0 +1,84 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+
+@available(iOS 13, *)
+extension ChannelControllerGeneric {
+    /// A wrapper object that exposes the controller variables in the form of `ObservableObject` to be used in SwiftUI.
+    public var observableObject: ObservableObject { .init(controller: self) }
+    
+    /// A wrapper object for `ChannelListController` type which makes it possible to use the controller comfortably in SwiftUI.
+    public class ObservableObject: SwiftUI.ObservableObject {
+        /// The underlying controller. You can still access it and call methods on it.
+        public let controller: ChannelControllerGeneric
+        
+        /// The channel matching the ChannelId.
+        @Published public private(set) var channel: ChannelModel<ExtraData>?
+        
+        ///
+        @Published public private(set) var messages: [MessageModel<ExtraData>] = []
+        
+        ///
+        @Published public private(set) var typingEvent: TypingEvent?
+        
+        ///
+        @Published public private(set) var memberEvent: MemberEvent?
```

Comment:
Events can't be exposed like this. They will be accessible via Combine publishers (CIS-313)

Reply:
I see, so should I just remove them from `SwiftUI `wrapper?

Reply:
Yep :)